src/Core/Framework/Plugin/PluginManagementService.php

```diff
@@ -60,12 +55,16 @@ public function extractPluginZip(string $file, bool $delete = true, ?string $sto
 
         if ($storeType) {
             $this->pluginExtractor->extract($archive, $delete, $storeType);
+            if ($storeType === self::PLUGIN) {
+                $this->cacheClearer->clearContainerCache();
```

Comment:
Can you add an test to cover this?

Reply:
I added tests for the whole cache clearing mechanic in the entire method now.